src/internal/connector/graph/http_wrapper.go

```diff
@@ -140,13 +140,20 @@ func defaultTransport() http.RoundTripper {
 }
 
 func internalMiddleware(cc *clientConfig) []khttp.Middleware {
-	return []khttp.Middleware{
+	mw := []khttp.Middleware{
 		&RetryMiddleware{
 			MaxRetries: cc.maxRetries,
 			Delay:      cc.minDelay,
 		},
+		khttp.NewRedirectHandler(),
 		&LoggingMiddleware{},
 		&ThrottleControlMiddleware{},
 		&MetricsMiddleware{},
 	}
+
+	if len(cc.appendMiddleware) > 0 {
```

Comment:
One limitation of `appendMiddleware` slice is that it can only be appended to the tail of middlewares. 
Is the longer term thought process here to supply the entire middleware chain through `appendMiddleware`?

Reply:
This alone feels pretty hacky and test oriented, and we have no need for additional middleware injection.  We definitely do not want to supply the entire middleware chain through this option, as that opens us up to all sorts of unexpected behaviors.  I'd strongly hesitate before extending this behavior at all.

Reply:
It would be good to add a small note next to `appendMiddleware` field description, saying that it's for test use only.